dt-assets/js/service-worker.js

```diff
@@ -31,109 +49,121 @@ addEventListener("message", (event) => {
-              switch(json[i].notification_name) {
-                case 'created':
-                  notify_title = 'New Contact Created';
-                  notify_body = 'A new contact was created and assigned to you.';
-                  break;
-                case 'assigned_to':
-                  notify_title = 'New Contact Assigned';
-                  notify_body = 'You have been assigned a new contact.';
-                  break;
-                case 'assigned_to_other':
-                  notify_title = 'Contact Reassigned';
-                  notify_body = 'A contact has been reassigned.';
-                  break;
-                case 'share':
-                  notify_title = 'Contact Shared';
-                  notify_body = 'A contact has been shared with you.';
-                  break;
-                case 'mention':
-                  notify_title = 'New Mention';
-                  notify_body = 'You were mentioned on a contact.';
-                  break;
-                case 'comment':
-                  notify_title = 'New Comment';
-                  notify_body = 'A new comment was left on a contact.';
-                  break;
-                case 'subassigned':
-                  notify_title = 'New Contact Subassigned';
-                  notify_body = 'A new contact has been subassigned to you.';
-                  break;
-                case 'milestone':
-                  notify_title = 'New Milestone';
-                  notify_body = 'A new milestone was added to a contact.';
-                  break;
-                case 'requires_update':
-                  notify_title = 'Update Required';
-                  notify_body = 'A contact requires an update.';
-                  break;
-                case 'contact_info_update':
-                  notify_title = 'Contact Updated';
-                  notify_body = 'A contact\'s details were modified.';
-                  break;
-                case 'assignment_declined':
-                  notify_title = 'User Declined Assignment';
-                  notify_body = 'A user declined assignment on a contact.';
-                  break;
-                default:
-                  // code
-              }
-              
-              // get window thing from pwa.js for icon
-              self.registration.showNotification(notify_title, {
-                body: notify_body,
-                icon: `${event.data["template_dir"]}/dt-assets/images/dt-caret.png`, // dt-assets/images/dt-caret.png
-                actions: [
-                  { action: 'open_link', title: 'Click here to open the link' }
-                ],
-                data: { link: notify_url }
-              });
+                // Extract post url from response data
+                const regex = /href="(.*)"/gm;
+                const matches = notification.notification_note.match(regex);
+                if (matches.length) {
+                  notify_url = matches[0];
+                }
```

Comment:
When I click "Open" on a notification, it is going to `https://dtplayground.local/href=%22https:/dtplayground.local/contacts/587`. I think it's using the wrong regex match. The first index is the whole match, including the starting href. I think if you use index 1 it should be just the URL that is within the parentheses of the regex pattern.